ads/google/adsense.js

```diff
@@ -31,12 +31,15 @@ export function adsense(global, data) {
       ['adClient', 'adSlot', 'adHost', 'adtest', 'tagOrigin', 'experimentId',
         'ampSlotIndex', 'adChannel', 'autoFormat', 'fullWidth']);
 
-  user().assert(
-      data['autoFormat'] != 'rspv'
-        || data['height'] == ADSENSE_RSPV_WHITELISTED_HEIGHT,
-      `Specified height ${data['height']} in <amp-ad> tag is not equal to ` +
+  if (data['autoFormat'] == 'rspv') {
+    user().assert(data.hasOwnProperty('fullWidth'),
```

Comment:
Out of curiousity, why not just force fullWidth here instead of throwing error?  I assume pub would prefer to get an ad.  Something like:
if (data['autoFormat'] == 'rsvp') {
    if (!data.HasOwnProperty('fullWidth')) {
       user().warn(TAG, 'Responsive AdSense ad units require full width, forcing attribute data-full-width');
       data['fullWidth'] = true;  // Or whatever value you expect
    }
}

Reply:
I think it's then past the point where the width is verified to be 100vw. So the publisher has manually removed the data-full-width, and changed the width to maybe 90vw instead of 100vw, and then here we put back the attribute without changing back the width. It's better that they see an actionable error message for tampering with the tag.

Reply:
FWIW for delayed fetch you can determine if the slot's width is already equal to viewport width by looking at window.context.initialIntersection.rootBounds.width == window.context.data.width and so you could elect to recover in that case.  I'll approve changes for now but wanted to bring it to your attention.